pcap-linux.c

```diff
@@ -2149,51 +1693,7 @@ pcap_stats_linux(pcap_t *handle, struct pcap_stat *stats)
 		*stats = handlep->stat;
 		return 0;
 	}
-	else
-	{
-		/*
-		 * If the error was EOPNOTSUPP, fall through, so that
-		 * if you build the library on a system with
-		 * "struct tpacket_stats" and run it on a system
-		 * that doesn't, it works as it does if the library
-		 * is built on a system without "struct tpacket_stats".
-		 */
-		if (errno != EOPNOTSUPP) {
-			pcap_fmt_errmsg_for_errno(handle->errbuf,
-			    PCAP_ERRBUF_SIZE, errno, "pcap_stats");
-			return -1;
-		}
-	}
-#endif
-	/*
-	 * On systems where the PACKET_STATISTICS "getsockopt()" argument
-	 * is not supported on PF_PACKET sockets:
-	 *
-	 *	"ps_recv" counts only packets that *passed* the filter,
-	 *	not packets that didn't pass the filter.  It does not
-	 *	count packets dropped because we ran out of buffer
-	 *	space.
-	 *
-	 *	"ps_drop" is not supported.
-	 *
-	 *	"ps_ifdrop" is supported. It will return the number
-	 *	of drops the interface reports in
-	 *	/sys/class/net/{if_name}/statistics/rx_{missed,fifo}_errors,
-	 *	if that is available.
-	 *
-	 *	"ps_recv" doesn't include packets not yet read from
-	 *	the kernel by libpcap.
-	 *
-	 * We maintain the count of packets processed by libpcap in
-	 * "handlep->packets_read", for reasons described in the comment
-	 * at the end of pcap_read_packet().  We have no idea how many
-	 * packets were dropped by the kernel buffers -- but we know
-	 * how many the interface dropped, so we can return that.
-	 */
-
-	stats->ps_recv = handlep->packets_read;
-	stats->ps_drop = 0;
-	stats->ps_ifdrop = handlep->stat.ps_ifdrop;
+	// MARIO: should indicate failure? If so, is it by returning -1?
```

Comment:
> should indicate failure?

Yes.

> If so, is it by returning -1?

Yes.  Also, set `handle->errbuf` to an appropriate error string, by calling `pcap_fmt_errmsg_for_errno()`.